src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -732,10 +732,26 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                 throw new InvalidOperationException("An 'id' must be specified in the host configuration.");
             }
 
-            JToken watchFiles = (JToken)config["watchFiles"];
```

Comment:
Renaming the existing watchFiles to fileWatchingEnabled (similar to fileLoggingEnabled)